src/energyplus/ForwardTranslator/ForwardTranslateCoilUserDefined.cpp

```diff
@@ -0,0 +1,141 @@
+
+    IdfObject idfObject(IddObjectType::Coil_UserDefined);
+
+    m_idfObjects.push_back(idfObject);
+
+    // Name
+    s = modelObject.name();
+    if (s) {
+      idfObject.setName(*s);
+    }
+
+    //PlantConnectionisUsed default value
+    idfObject.setString(Coil_UserDefinedFields::PlantConnectionisUsed, "No");
+
+    // PlantConnectionInletNodeName
+    bool plantUsed = false;
+    if (boost::optional<ModelObject> mo = modelObject.waterInletModelObject()) {
+      if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+        idfObject.setString(Coil_UserDefinedFields::PlantConnectionInletNodeName, node->name().get());
+        //PlantConnectionisUsed to Yes since Node is connected
+        idfObject.setString(Coil_UserDefinedFields::PlantConnectionisUsed, "Yes");
+        plantUsed = true;
+      }
+    }
+
+    // PlantConnectionOutletNodeName
+
+    if (boost::optional<ModelObject> mo = modelObject.waterOutletModelObject()) {
+      if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+        idfObject.setString(Coil_UserDefinedFields::PlantConnectionOutletNodeName, node->name().get());
+      }
+    }
+    // remove plant actuators if no plant connections
+    if (!plantUsed) {
+      modelObject.plantMinimumMassFlowRateActuator().get().remove();
+      modelObject.plantMaximumMassFlowRateActuator().get().remove();
+      modelObject.plantDesignVolumeFlowRateActuator().get().remove();
+      modelObject.plantMassFlowRateActuator().get().remove();
+      modelObject.plantOutletTemperatureActuator().get().remove();
+    }
+    // AirConnection1InletNodeName
+
+    if (boost::optional<ModelObject> mo = modelObject.airInletModelObject()) {
+      if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+        idfObject.setString(Coil_UserDefinedFields::AirConnection1InletNodeName, node->name().get());
+      }
+    }
+
+    // AirConnection1OutletNodeName
+
+    if (boost::optional<ModelObject> mo = modelObject.airOutletModelObject()) {
+      if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+        idfObject.setString(Coil_UserDefinedFields::AirConnection1OutletNodeName, node->name().get());
+      }
+    }
+
+    // NumberofAirConnections
+    // value is 1 since its a WaterToAirComponent
+    idfObject.setInt(Coil_UserDefinedFields::NumberofAirConnections, 1);
+
+    // OverallModelSimulationProgramCallingManagerName
+
+    if (boost::optional<EnergyManagementSystemProgramCallingManager> pcm = modelObject.overallModelSimulationProgramCallingManager()) {
+      idfObject.setString(Coil_UserDefinedFields::OverallModelSimulationProgramCallingManagerName, pcm->name().get());
+    }
+
+    // ModelSetupandSizingProgramCallingManagerName
+
+    if (boost::optional<EnergyManagementSystemProgramCallingManager> pcm = modelObject.modelSetupandSizingProgramCallingManager()) {
+      idfObject.setString(Coil_UserDefinedFields::ModelSetupandSizingProgramCallingManagerName, pcm->name().get());
+    }
```

Comment:
I guess the EMS objects are all in the Always Translate list in the FT...? I think it'll still advocate that for all EMS children of this object I'd want to see a translateAndMapModelObject(child)...